### PR TITLE
added CartItem and ShoppingCart to project

### PR DESCRIPTION
**Pull request by vova5179**

### CODE REVIEW

File: src/main/java/mate/academy/booksstore/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        this.shoppingCartService = shoppingCartService;
+    }
+
+    @PostMapping()
+    @ResponseStatus(HttpStatus.CREATED)
+    @PreAuthorize("hasRole('USER') or hasRole('ADMIN')")
+    @Operation(summary = "Add book to shopping cart")
+    public ShoppingCartDto addBookToShoppingCart(@RequestBody @Valid
+                                                     CartItemRequestDto requestDto) {
+        return shoppingCartService.addCartItemToShoppingCart(requestDto);
```
Comment (Rommelua) [inline src/main/java/mate/academy/booksstore/controller/ShoppingCartController.java]: Use this approach for all controller methods.
```suggestion
    User user = (User) authentication.getPrincipal();
    return shoppingCartService.addToCart(request, user.getId());
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/booksstore/dto/CartItemRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package mate.academy.booksstore.dto;
+
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+
+@Data
+public class CartItemRequestDto {
+    @NotNull
```
Comment (koval-kn) [inline src/main/java/mate/academy/booksstore/dto/CartItemRequestDto.java:9]: it's better to ident annotated fields:
```suggestion
    @NotNull
    private Long bookId;
    
    @Min(1)
    private Integer quantity;
```

File: src/main/java/mate/academy/booksstore/mapper/ShoppingCartMapper.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import mate.academy.booksstore.config.MapperConfig;
+import mate.academy.booksstore.dto.ShoppingCartDto;
+import mate.academy.booksstore.model.ShoppingCart;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(config = MapperConfig.class, uses = CartItemMapper.class)
+public interface ShoppingCartMapper {
+
```
Comment (koval-kn) [inline src/main/java/mate/academy/booksstore/mapper/ShoppingCartMapper.java]: redundant blank line

File: src/main/java/mate/academy/booksstore/repository/ShoppingCartRepository.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package mate.academy.booksstore.repository;
+
+import java.util.Optional;
+import mate.academy.booksstore.model.ShoppingCart;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+
```
Comment (koval-kn) [inline src/main/java/mate/academy/booksstore/repository/ShoppingCartRepository.java]: redundant blank line

**Review by koval-kn — CHANGES_REQUESTED** [review]
needs minor changes

**Review by koval-kn — CHANGES_REQUESTED** [review]
comments was not fixed, see the last ones

**Pull request merged**